lib/src/mt_ptp.c

```diff
@@ -277,6 +299,24 @@ static inline int ptp_timesync_read_rx_time(struct mt_ptp_impl* ptp, uint32_t fl
   return ret;
 }
 
+#ifdef MTL_HAS_DPDK_TIMESYNC_ADJUST_FREQ
+static inline int ptp_timesync_adjust_freq(struct mt_ptp_impl* ptp, int64_t ppm,
+                                           int64_t delta) {
+  int ret;
+
+  if (ptp->no_timesync) {
+    ptp_no_timesync_adjust(ptp, delta);
+    return 0;
+  }
+
+  ptp_timesync_lock(ptp);
+  ret = rte_eth_timesync_adjust_freq(ptp->port_id, ppm);
```

Comment:
Should we call `ptp_timesync_adjust_time` as the fallback path if `rte_eth_timesync_adjust_freq` is failed to run? Ex, on the standard E810 nic.

Reply:
Done.